src/Illuminate/Database/Eloquent/Relations/BelongsToMany.php

```diff
@@ -131,7 +131,7 @@ public function addConstraints()
      * @param  \Illuminate\Database\Eloquent\Builder|null  $query
      * @return $this
      */
-    protected function performJoin($query = null)
+    protected function performJoin(Builder $query = null)
```

Comment:
Please only make the changes strictly needed for this PR. Please send that typehint change to the master branch, since it's breaking.